pkg/mover/intermediate.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+package mover
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"github.com/vmware-tanzu/asset-relocation-tool-for-kubernetes/internal"
+	"helm.sh/helm/v3/pkg/chart"
+)
+
+var (
+	// ErrNotIntermediateBundle when a verified path does not have expected intermediate bundle contents
+	ErrNotIntermediateBundle = errors.New("not an intermediate chart bundle")
+)
+
+func saveIntermediateBundle(cd *ChartData, tarFile string, log Logger) error {
+	tfw, err := newTarFileWriter(tarFile)
+	if err != nil {
+		return err
+	}
+
+	// hints file goes first to be extracted quickly on demand
```

Comment:
From your comment. How does writing order affect the extraction times?

Reply:
The hints file is a small file that we load fully in memory. It does not make sense to add it last or in the middle of the tar. the tar is always accessed sequentially, anything you want to be found early, with little iterations, must go first.

The hints file is the tiniest of the 3 components, so it should go first. The chart needs to be extracted, but we do not want to extract the images with it, so we can just put is before the images and let it skip the rest. Images are the biggest in terms of time due to layer IO, so seek time is a bit less important there IMHO.